src/service/document-click.js

```diff
@@ -129,6 +135,43 @@ export class ClickHandler {
     }
   }
 
+  /**
+   * Navigates a window to a URL.
+   *
+   * If opt_requestedBy matches a feature name in a <meta> tag with attribute
+   * name="amp-to-amp-navigation", then treats the URL as an AMP URL (A2A).
+   *
+   * @param {!Window} win
+   * @param {string} url
+   * @param {string=} opt_requestedBy
+   */
+  navigateTo(win, url, opt_requestedBy) {
+    if (!isProtocolValid(url)) {
+      user().error(TAG, 'Cannot navigate to invalid protocol: ' + url);
+      return;
+    }
+
+    // If this redirect was requested by a feature that opted into A2A,
+    // try to ask the viewer to navigate this AMP URL.
+    if (opt_requestedBy) {
+      if (!this.a2aFeatures_) {
+        const meta = this.rootNode_.querySelector(
+            'meta[name="amp-to-amp-navigation"]');
+        this.a2aFeatures_ = (meta && meta.hasAttribute('content'))
+          ? meta.getAttribute('content').split(',').map(s => s.trim())
```

Comment:
We can remove the `#map` by splitting whitespace: `content.trim().split(/\s*,\s*/)`

Reply:
I'd prefer to leave as is for readability.